eureka2-client/src/main/java/com/netflix/eureka2/client/registration/EurekaRegistrationClientImpl.java

```diff
@@ -84,6 +88,6 @@ public void call() {
     @Override
     public void shutdown() {
         logger.info("Shutting down RegistrationClient");
-        // nothing to shutdown
+        shutdownSubject.onCompleted();
```

Comment:
Let's guard this with an atomic state boolean to observe the rx contract and not call onCompleted more than once.